packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -64,84 +67,82 @@ describe('Model > SubjectStore', function () {
       })
 
       it('should request more subjects', function () {
-        expect(populateSpy).to.have.been.calledOnce()
+        // Once for initialization and once after the queue has been advanced to less than 3 subjects
+        expect(rootStore.subjects.populateQueue).to.have.been.calledTwice()
+      })
+
+      describe('when the initial response has less than three subjects', function () {
+        let rootStore
+        before(function () {
+          const clientStub = stubPanoptesJs({ workflows: workflow, subjects: shortListSubjects })
+          rootStore = setupStores(clientStub)
+        })
+
+        it('should request more subjects', function () {
+          // Once for initialization and again since less than three subjects in initial response
+          expect(rootStore.subjects.populateQueue).to.have.been.calledTwice()
+        })
+      })
+
+      describe('when the initial response has no subjects', function () {
+        let rootStore
+        before(function () {
+          const clientStub = stubPanoptesJs({ workflows: workflow, subjects: [] })
+          rootStore = setupStores(clientStub)
+        })
+
+        it('should request more subjects', function () {
+          // Once for initialization
+          expect(rootStore.subjects.populateQueue).to.have.been.calledOnce()
+        })
       })
     })
 
     describe('after emptying the queue', function () {
```

Comment:
```suggestion
    describe('when no subjects are returned from the API', function () {
```
In retrospect, this is a better description of this test case.

Reply:
We're testing the initial API response having no subjects in the previous describe block. I think the confusion here happened with a bad panoptes js client stub setup.

Reply:
I intended this test to test the case where the queue is being advanced but isn’t being refilled from Panoptes.